ignite/metrics/metric.py

```diff
@@ -292,6 +292,13 @@ def __getitem__(self, index: Any):
 
         return MetricsLambda(lambda x: x[index], self)
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
```

Comment:
Maybe we can simply return the dict without copying it ?

Reply:
I try that